### PR TITLE
Keypair and keyless signing methods

### PR DESCRIPTION
**Pull request by javuto**
Defaulting to `keypair` signing method until being able to fix the timeout for `keyless`

### CODE REVIEW

File: .github/actions/build-sign-publish-chainlink/action.yml [hunk 1]
```diff
@@ -160,35 +168,66 @@ runs:
...
-  - if: inputs.sign-images == 'true'
-    name: Sign the published root Docker image
+  - if: inputs.sign-images == 'true' && inputs.sign-method == 'keypair'
+    name: Sign the published root Docker image using keypair method
+    shell: sh
+    env:
+      COSIGN_PASSWORD: "${{ inputs.cosign-password }}"
+    run: |
+      echo "${{ inputs.cosign-private-key }}" > cosign.key
+      cosign sign --key cosign.key "${{ env.root_image_name }}"
```
Comment (chainchad) [inline .github/actions/build-sign-publish-chainlink/action.yml:178]: After this step, would it be safer to remove the key so it never gets accidentally saved in an artifact?

```suggestion
      cosign sign --key cosign.key "${{ env.root_image_name }}"
      rm -f cosign.key
```

File: .github/actions/build-sign-publish-chainlink/action.yml [hunk 2]
```diff
@@ -160,35 +168,66 @@ runs:
...
+
+  - if: inputs.verify-signature == 'true' && inputs.sign-method == 'keypair'
+    name: Verify the signature of the published root Docker image using keypair
+    shell: sh
+    run: |
+      echo "${{ inputs.cosign-public-key }}" > cosign.key
+      cosign verify --key cosign.key "${{ env.root_image_name }}"
```
Comment (chainchad) [inline .github/actions/build-sign-publish-chainlink/action.yml:185]:
```suggestion
      cosign verify --key cosign.key "${{ env.root_image_name }}"
      rm -f cosign.key
```

File: .github/actions/build-sign-publish-chainlink/action.yml [hunk 3]
```diff
@@ -160,35 +168,66 @@ runs:
...
-  - if: inputs.sign-images == 'true'
-    name: Sign the published non-root Docker image
+  - if: inputs.sign-images == 'true' && inputs.sign-method == 'keypair'
+    name: Sign the published non-root Docker image using keypair method
+    shell: sh
+    env:
+      COSIGN_PASSWORD: "${{ inputs.cosign-password }}"
+    run: |
+      echo "${{ inputs.cosign-public-key }}" > cosign.key
+      cosign sign "${{ env.nonroot_image_name }}"
```
Comment (chainchad) [inline .github/actions/build-sign-publish-chainlink/action.yml:210]:
```suggestion
      cosign sign "${{ env.nonroot_image_name }}"
      rm -f cosign.key
```

File: .github/actions/build-sign-publish-chainlink/action.yml [hunk 4]
```diff
@@ -160,35 +168,66 @@ runs:
...
+
+  - if: inputs.verify-signature == 'true' && inputs.sign-method == 'keypair'
+    name: Verify the signature of the published non-root Docker image using keypair
+    shell: sh
+    run: |
+      echo "${{ inputs.cosign-public-key }}" > cosign.key
+      cosign verify --key cosign.key "${{ env.nonroot_image_name }}"
```
Comment (chainchad) [inline .github/actions/build-sign-publish-chainlink/action.yml:217]:
```suggestion
      cosign verify --key cosign.key "${{ env.nonroot_image_name }}"
      rm -f cosign.key
```

**Review by chainchad — COMMENTED** [review]
Thoughts on removing the key after each step?

**Review by chainchad — APPROVED** [review]
👍 (after removing the key if you think it's needed)

**Review by chainchad — APPROVED** [review]

File: .github/actions/build-sign-publish-chainlink/action.yml [hunk 5]
```diff
@@ -38,6 +38,15 @@ inputs:
     description: When set to the string boolean value of "true", the resulting build image will be signed
     default: "false"
     required: false
+  cosign-private-key:
+    description: The private key to be used with cosign to sign the image
+    required: false
+  cosign-password:
+    description: The password to decrypt the cosign private key needed to sign the image
+  sign-method:
```
Comment (HenryNguyen5) [inline .github/actions/build-sign-publish-chainlink/action.yml:46]: should we prefix all cosign related inputs with `cosign`?

Comment (javuto) [reply .github/actions/build-sign-publish-chainlink/action.yml:46]: Good idea, let me draft up another PR, that uses a newer version of cosign and does this.

**Pull request merged**